pyp2rpm/package_getters.py

```diff
@@ -128,21 +135,22 @@ def __init__(self, client, name, version=None, save_dir=None):
             sys.stderr.write("Failed to connect to server: {0} \n".format(e))
             raise SystemExit(3)
 
-        if not self.versions:  # If versions is empty list then there is no such package on PyPI
+        # If versions is empty list then there is no such package on PyPI
+        if not self.versions:
+            logger.error('Package "{0}" could not be found on PyPI.'.format(
+                name))
             raise exceptions.NoSuchPackageException(
                 'Package "{0}" could not be found on PyPI.'.format(name))
-            logger.error('Package "{0}" could not be found on PyPI.'.format(name))
 
         self.version = version or self.versions[0]
 
         # if version is specified, will check if such version exists
         if version and self.client.release_urls(name, version) == []:
             raise exceptions.NoSuchPackageException(
-                'Package with name "{0}" and version "{1}" could not be found on PyPI.'.format(
-                    name, version))
-            logger.error(
-                'Package with name "{0}" and version "{1}" could not be found on PyPI.'.format(
-                    name, version))
+                'Package with name "{0}" and version "{1}" could not be '
```

Comment:
I've realized this message won't ever be written into log, because an exception is raised one line before :) Exception appears in the log so I think it is not necessary to duplicate the message.

Reply:
Oh, right :)